doc/md/tutorial-todo-gql-node.md

```diff
@@ -0,0 +1,156 @@
+Node interface, read the following paragraphs that were taken from [relay.dev](https://relay.dev/graphql/objectidentification.htm#sel-DABDDBAADLA0Cl0c):
+
+> To provide options for GraphQL clients to elegantly handle for caching and data refetching GraphQL servers need to expose object identifiers in a standardized way. In the query, the schema should provide a standard mechanism for asking for an object by ID. In the response, the schema provides a standard way of providing these IDs.
+>
+> We refer to objects with identifiers as “nodes”. An example of both of those is the following query:
+>
+>  ```graphql
+>   {
+>       node(id: "4") {
+>           id
+>          ... on User {
+>               name
+>           }
+>       }
+>   }
+> ```
+
+#### Clone the code (optional)
+
+The code for this tutorial is available under [github.com/a8m/ent-graphql-example](https://github.com/a8m/ent-graphql-example), 
+and tagged (using Git) in each step. If you want to skip the basic setup and start with the initial version of the GraphQL
+server, you can clone the repository and checkout `v0.1.0` as follows:
+
+```console
+git clone git@github.com:a8m/ent-graphql-example.git
+cd ent-graphql-example 
+go run ./cmd/todo/
+```
+
+## Implementation
+
+Ent supports the Node interface with its GraphQL integration, and the steps for adding this to your application are 
+very simple. We start by adding the `Node` interface to our GraphQL schema:
+
+```diff
++interface Node {
++    id: ID!
++}
+
+-type Todo {
++type Todo implements Node {
+    id: ID!
+    createdAt: Time
+    status: Status!
+    priority: Int!
+    text: String!
+    parent: Todo
+    children: [Todo!]
+}
+
+type Query {
+    todos: [Todo!]
++   node(id: ID!): Node
++   nodes(ids: [ID!]!): [Node]!
+}
+```
+
+Then, we tell gqlgen that Ent provides this interface by editing the `gqlgen.yaml` file as follows:
+
+```diff
+# This section declares type mapping between the GraphQL and Go type systems.
+models:
+  # Defines the ID field as Go 'int'.
+  ID:
+    model:
+      - github.com/99designs/gqlgen/graphql.IntID
++ Node:
++   model:
++     - todo/ent.Noder
+  Status:
+    model:
+      - todo/ent/todo.Status
+```
+
+As you can see, this is a 7 lines change. Let's run the code generation using:
+
+```console
+go generate ./...
+```
+
+Like before, we need to implement the GraphQL resolve in the `todo.resolvers.go` file, but that's simple.
+Let's replace the default resolvers with the following:
+
+```go
+func (r *queryResolver) Node(ctx context.Context, id int) (ent.Noder, error) {
+	return r.client.Noder(ctx, id)
+}
+
+func (r *queryResolver) Nodes(ctx context.Context, ids []int) ([]ent.Noder, error) {
+	return r.client.Noders(ctx, ids)
+}
+```
+
+## Query Nodes
+
+Now, we're ready to test our new GraphQL resolvers. Let's start with creating a few todo items by running this
+query multiple times (changing variables is optional):
+
+```graphql
+mutation CreateTodo($todo: TodoInput!) {
```

Comment:
Is something missing here? This just works? Where do the values come from?

Reply:
I needed to paste here the parameters/variables to the mutation as well. Fixed

Reply:
See the "Query Variables" in the comment below.